test/src/unit-sparse-unordered-with-dups-reader.cc

```diff
@@ -1518,18 +1464,55 @@ TEST_CASE_METHOD(
     expected_var_buffer_size = 14;
   }
 
-  compute_var_size_offsets_test(
-      var_buffer_size,
-      bitmaps,
-      capacity,
-      num_tiles,
-      first_tile_min_pos,
-      offsets_buffer,
-      cell_offsets,
-      expected_buffers_full,
-      expected_cell_offsets,
-      expected_result_tiles_size,
-      expected_var_buffer_size);
+  auto&& [array, fragments] = open_default_array_1d_with_fragments(capacity);
+
+  // Make a vector of tiles.
+  std::vector<UnorderedWithDupsResultTile<uint64_t>> rt;
+  for (uint64_t t = 0; t < num_tiles; t++) {
+    rt.emplace_back(0, t, *fragments[0]);
+
+    // Allocate and set the bitmap if required.
+    if (bitmaps[t].size() > 0) {
+      rt.back().bitmap() = bitmaps[t];
+      rt.back().count_cells();
+    }
+  }
+
+  // Create the result_tiles pointer vector.
```

Comment:
Refactoring note: anywhere that we write "create the <thing>" and do some manual loop is a place where we might need a class.